### DISCUSSION TITLE
Training season when joining a crew

### DISCUSSION THREAD

**Discussion by hweimer** [raceintospace/raceintospace#573] [Ideas]
Right now, you can reasonably fly your first manned mission in the fall of 1959, which is a bit too early when comparing to historical standards. One way to push this back is to reintroduce the one season of training for 'nauts joining a crew, but only once per program, so you don't have to start over when a flt crew is broken up.

**Comment by peyre** [comment]
I see the issue you're trying to solve; that's probably a valid concern.  So you're thinking that once only (say, for one turn only) you
would have to wait a turn after assigning people to crews?  That's an innovative idea.  I could see it being a little tricky to explain to players, but we could get around that.

Or perhaps we could require the one-turn wait anytime before, say, 1962, and change the Help item to say - instead of "you have to wait a turn before assigning to a mission", something more like "Before 1962, the details of crewed spaceflight are still being worked out, requiring crews to spend a turn training before being assigned to a mission".

**Reply by hweimer** [reply]
> So you're thinking that once only (say, for one turn only) you
would have to wait a turn after assigning people to crews?

Yeah, but once per program and astronaut. So if you have someone who had undergone training already, they would not have to wait a second time after joining a new crew.

> I could see it being a little tricky to explain to players, but we could get around that.

You could have a pop-up showing a help text with something like "This flight crew contains a member who has not been assigned to the program before and therefore needs one season of training to become familiar with it. You can assign this crew to a mission starting with the next turn."

**Reply by peyre** [reply]
That would be a whole extra set of data for the game to keep track of (or an extra attribute for every astronaut), but I could see it working.  It would certainly be more fair than in the original game.  In BARIS, even if you broke a crew and immediately recreated it with the same people in the same places, it still had to wait a turn before being assigned to a mission.

**Reply by hweimer** [reply]
You are absolutely right that this would require to store additional information fore each 'naut, but I guess it can be done. One possibility would be to wait for #478 (I know, my standard response these days). The other would be to encode this into Astros::TrainingLevel. Current;ly, only 3 bits are used to keep track of the training status, so one could use the 5 highest significant bits to store a flag whether a 'naut has already done the one season of additional training for each program. For example, the highest bit would encode whether someone has already trained on Mercury/Vostok, the second highest bit for Gemini/Voskhod, and so on.

**Reply by peyre** [reply]
So what _is_ the holdup for 478?  (Besides the conflict with data.h of course.)

**Reply by hweimer** [reply]
Per https://github.com/raceintospace/raceintospace/pull/478#issuecomment-720154466, it's deferred until after the next release.

**Reply by peyre** [reply]
Oh yes!  

Speaking of which, why don't we go ahead with a release?

At the least, maybe we could put together a prerelease so people can get a taste of what's in the works.

**Reply by hweimer** [reply]
As far as I can see, there is still the question surrounding the license for the music files, see #420. But I am sympathetic towards your approach to keep the same licensing as in 1.1, since @mkmccarty appears to have approved it back then.

**Reply by peyre** [reply]
Thanks, that's my thinking exactly.

**Comment by mkmccarty** [comment]
I spoke with Fritz on Friday about the music.  He feels that the MIDI files
are okay to be used freely.  The Brian Langsbard orchestration CD audio
shouldn't be freely used.  I never release those with the original
source+data release.

Michael

**Reply by peyre** [reply]
Ah, too bad; I do love that music.  But the important thing's to get the licensing sorted out; thanks for doing that Michael!  And it should be reasonably easy to find free substitutes.  I'm just glad we're ok to keep the MIDIs.

**Reply by peyre** [reply]
I assume we couldn't use the old MIDI intro and end music from the floppy version?

**Reply by hweimer** [reply]
The intro music being played in the game *is* derived from the MIDI files (using a software synthesizer, I assume). The CVS repository had the MIDI files included, so maybe it would be a good idea to add them back, in case someone wants to come up with better .ogg versions. The tracks from the CD version were never referenced anywhere in the game, so I'm not sure they were even included in any of the previous releases.

**Comment by hweimer** [comment]
Fantastic, thank you for sorting this out!

**Comment by mkmccarty** [comment]
To be clear, those MIDI files can be used with the game, not for other
things unrelated to RIS.  If there is a desire to use them for other
projects then Fritz should be contacted for arrangements.

**Reply by peyre** [reply]
That makes perfect sense.

**Comment by hweimer** [comment]
Am 29. August 2021 21:25:46 MESZ schrieb Michael K McCarty ***@***.***>:
> To be clear, those MIDI files can be used with the game, not for other
> things unrelated to RIS.  If there is a desire to use them for other
> projects then Fritz should be contacted for arrangements.
> 
> On Sun, Aug 29, 2021 at 12:15 PM Hendrik W. ***@***.***>
> wrote:
> 
> > The intro music being played in the game *is* derived from the MIDI
> files
> > (using a software synthesizer, I assume). The CVS repository had the
> MIDI
> > files included, so maybe it would be a good idea to add them back,
> in case
> > someone wants to come up with better .ogg versions. The tracks from
> the CD
> > version were never referenced anywhere in the game, so I'm not sure
> they
> > were even included in any of the previous releases.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> >
> <https://github.com/raceintospace/raceintospace/discussions/573#discussioncomment-1250962>,
> > or unsubscribe
> >
> <https://github.com/notifications/unsubscribe-auth/AAAOCHS4COEWNZ6HCTF3W3DT7KBNZANCNFSM5CTHVJWA>
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> >
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> > or Android
> >
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
> >
> >

Thank you for the clarification, that means the MIDI files cannot be distributed under the GPL. But I assume that for the rest of the game's data (i.e., graphics and videos), the GPL is fine?

**Comment by mkmccarty** [comment]
Yeah, GPL for the other data is good.
